### PR TITLE
Advanced course opportunity

### PR DESCRIPTION
**Pull request by komalahire**
Created advanced course opportunity page

### CODE REVIEW

File: src/pages/Opportunities/index.js [hunk 1]
```diff
@@ -1,30 +1,16 @@
...
-      headers: {
-        accept: "application/json",
-        Authorization: Token,
-      },
-    }).then((res) => {
-      setUsers(res.data.user.partner_id);
-    });
-  }, []);
+  const url = window.location.href;
+  let partnerIdFromAndroid = url.split("partner_id=");
```
Comment (pysaquib) [inline src/pages/Opportunities/index.js]: If you split it by `partner_id=` it doesn't guarantee that the element on position 1 will always be the `id`
For example, 
`http://merakilearn.org/login?partner_id=65` will work
BUT
`http://merakilearn.org/login?partner_id=65&referrer=breakthrough` will NOT
Here `partnerIdFromAndroid = 65&referrer=breakthrough`

File: src/pages/Opportunities/index.js [hunk 2]
```diff
@@ -35,17 +21,16 @@ function Opportunities() {
         Authorization: user.data.token,
       },
     }).then((res) => {
-      // console.log(res, '===========')
       setPartner(res.data.partners);
     });
   }, []);
 
   let slug;
   for (const item of partner) {
```
Comment (pysaquib) [inline src/pages/Opportunities/index.js:29]: Why not use _.filter?

File: src/pages/Opportunities/index.js [hunk 3]
```diff
@@ -35,17 +21,16 @@ function Opportunities() {
...
-    if (item.id == userId) {
+    if (item.id == partnerId) {
```
Comment (pysaquib) [inline src/pages/Opportunities/index.js]: Even if you're iterating through the partners array two if conditions can be combined into one
```
     if (item.id == partnerId || item.id == partnerIdFromAndroid[1]) {
            slug = item.slug
      }
```

**Review by pysaquib — CHANGES_REQUESTED** [review]
Refer to comments

File: src/pages/Opportunities/index.js [hunk 4]
```diff
@@ -1,7 +1,58 @@
...
+import { METHODS } from "../../services/api";
 
 function Opportunities() {
+  const [partner, setPartner] = useState([]);
+  const [users, setUsers] = useState();
+  const user = useSelector(({ User }) => User);
+  const userId = user.data.user.partner_id;
+
+  const Token =
+    "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjEzMTYiLCJlbWFpbCI6InBvb25hbUBuYXZndXJ1a3VsLm9yZyIsImlhdCI6MTYzNDU0ODc1OCwiZXhwIjoxNjY2MTA2MzU4fQ.PbvSCQbum3kigzwJs7BoSJPOFrqx6F70Rj6c-5AWOOo";
```
Comment (pysaquib) [inline src/pages/Opportunities/index.js]: Why is this not in env variable?

**Review by pysaquib — CHANGES_REQUESTED** [review]
Check comments

File: src/pages/Opportunities/index.js (hunk 1 above)
Comment (komalahire) [reply src/pages/Opportunities/index.js]: So the android team is sending partner id  in this format

https://www.merakilearn.org/opportunities?partner_id=value

Comment (nilshah98) [reply src/pages/Opportunities/index.js]: @komalahire but in the future, they can send more parameters.
It is better to add an extra check as @pysaquib suggested.
It should not be a very tough task.

**Pull request merged**